From 82c44052600bc460129f97198c77cde92935530b Mon Sep 17 00:00:00 2001
From: No767 <73260931+No767@users.noreply.github.com>
Date: Fri, 21 Jan 2022 00:00:31 -0800
Subject: [PATCH 2/6] Upgrade Docker Image to Python 3.10.2

---
 Dockerfile   | 2 +-
 changelog.md | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Dockerfile b/Dockerfile
index 1e2705c..426d3d7 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM python:3.10.1-bullseye
+FROM python:3.10.2-bullseye
 WORKDIR /Bot
 COPY Pipfile ./ /Bot/
 COPY Pipfile.lock ./ /Bot/
diff --git a/changelog.md b/changelog.md
index 83c3b4d..3068a38 100644
--- a/changelog.md
+++ b/changelog.md
@@ -3,3 +3,4 @@
 
 # Changes
 - Fix DeviantArt Token Refresher (the insert would not make it through)
+- Update Docker Image to Python 3.10.2
-- 
2.34.1

