From c6f2311b4e80512b64546d70caa5ec49b27fa636 Mon Sep 17 00:00:00 2001
From: No767 <73260931+No767@users.noreply.github.com>
Date: Fri, 11 Mar 2022 19:30:15 -0800
Subject: [PATCH] Fix Duplicate Fields for Modrinth Cog

---
 Bot/Cogs/modrinth.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Bot/Cogs/modrinth.py b/Bot/Cogs/modrinth.py
index fed7675..4649c39 100644
--- a/Bot/Cogs/modrinth.py
+++ b/Bot/Cogs/modrinth.py
@@ -161,17 +161,18 @@ class ModrinthV3(commands.Cog):
                             if keys not in versionFilter:
                                 embedVar.add_field(
                                     name=keys, value=value, inline=True)
-                                embedVar.remove_field(-8)
+                                embedVar.remove_field(-14)
                         for fileItems in dictVersions["files"]:
                             for k, v in fileItems.items():
                                 if k not in "hashes":
                                     embedVar.add_field(
                                         name=k, value=v, inline=True)
+                                    embedVar.remove_field(-14)
                             for hashKey, hashValue in fileItems["hashes"].items():
                                 embedVar.add_field(
                                     name=hashKey, value=hashValue, inline=True
                                 )
-                        embedVar.remove_field(-8)
+                                embedVar.remove_field(-14)
                         embedVar.title = dictVersions["name"]
                         embedVar.description = dictVersions["changelog"]
                         await ctx.respond(embed=embedVar)
-- 
2.35.1

