From 015f4da6b78ca0a45390d3088b0c499adcc139b5 Mon Sep 17 00:00:00 2001
From: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Date: Mon, 10 Jan 2022 00:55:09 +0000
Subject: [PATCH] Format All Cogs

---
 Bot/Cogs/reddit.py | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/Bot/Cogs/reddit.py b/Bot/Cogs/reddit.py
index 11cb4b2..c646ced 100644
--- a/Bot/Cogs/reddit.py
+++ b/Bot/Cogs/reddit.py
@@ -1,8 +1,8 @@
 import os
 import random
 
-import discord
 import asyncpraw
+import discord
 from discord.ext import commands
 from dotenv import load_dotenv
 
@@ -56,10 +56,14 @@ class reddit(commands.Cog):
 
     @commands.command(name="reddit", help="browses on reddit")
     async def reddit(self, ctx, *, search: str):
-        async with asyncpraw.Reddit(client_id=Reddit_ID, client_secret=Reddit_Secret, user_agent="ubuntu:kumiko:v0.1.0 (by /u/No767)") as api:
+        async with asyncpraw.Reddit(
+            client_id=Reddit_ID,
+            client_secret=Reddit_Secret,
+            user_agent="ubuntu:kumiko:v0.1.0 (by /u/No767)",
+        ) as api:
             original_search = search
             try:
-                if "r/" in search: 
+                if "r/" in search:
                     search = search.split("/")
                     sub = search[1]
                     search = "all"
@@ -71,9 +75,9 @@ class reddit(commands.Cog):
                     post
                     async for post in searcher
                     if ".jpg" in post.url
-                       or ".png" in post.url
-                       or ".gif" in post.url
-                       and not post.over_18
+                    or ".png" in post.url
+                    or ".gif" in post.url
+                    and not post.over_18
                 ]
                 post = random.choice(posts)
                 submission = post
-- 
2.33.1.windows.1

