From e21c0a4b901002206f3a06b1b458a644f4583351 Mon Sep 17 00:00:00 2001
From: No767 <73260931+No767@users.noreply.github.com>
Date: Mon, 14 Mar 2022 00:23:31 -0700
Subject: [PATCH] Update RinHelp with Reddit Cmds

---
 Bot/Cogs/rinhelp.py | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/Bot/Cogs/rinhelp.py b/Bot/Cogs/rinhelp.py
index 4c370a7..8a85a5f 100644
--- a/Bot/Cogs/rinhelp.py
+++ b/Bot/Cogs/rinhelp.py
@@ -140,16 +140,12 @@ class rinhelp(commands.Cog):
                 embedVar.add_field(
                     name="`reddit`", value="searches on reddit", inline=True
                 )
-                embedVar.add_field(
-                    name="`transmeme`",
-                    value="searches on reddit that include trans and other LGBTQ+ subreddits",
-                    inline=True,
-                )
-                embedVar.add_field(
-                    name="`meme`",
-                    value="searches on reddit that include defined search topics regarding memes",
-                    inline=True,
-                )
+                embedVar.add_field(name="`reddit-new`", value="Returns 5 new posts from any subreddit", inline=True)
+                embedVar.add_field(name="`reddit-top`", value="Returns 5 top posts from any subreddit", inline=True)
+                embedVar.add_field(name="`reddit-hot`", value="Returns 5 hot posts from any subreddit", inline=True)
+                embedVar.add_field(name="`reddit-comemnts`", value="Returns up to 10 comemnts from a given post ID", inline=True)
+                embedVar.add_field(name="`reddit-user`", value="Provides info about the given Redditor", inline=True)
+                embedVar.add_field(name="`reddit-user-comments`", value="Returns up to 10 comments from a given Redditor", inline=True)
                 embedVar.set_author(
                     name="Rin Help - Reddit", icon_url=bot.user.display_avatar
                 )
-- 
2.35.1

