From 7eb532dede4244e7d365f087066b429804bab45d Mon Sep 17 00:00:00 2001
From: No767 <73260931+No767@users.noreply.github.com>
Date: Sun, 27 Mar 2022 22:18:10 -0700
Subject: [PATCH] Use Different PostgreSQL DB For DeviantArt Cog

---
 Bot/Cogs/deviantart.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Bot/Cogs/deviantart.py b/Bot/Cogs/deviantart.py
index 6312dfc..f715e14 100644
--- a/Bot/Cogs/deviantart.py
+++ b/Bot/Cogs/deviantart.py
@@ -25,7 +25,7 @@ class tokenFetcher:
     async def get(self):
         meta = MetaData()
         engine = create_async_engine(
-            f"postgresql+asyncpg://{Username}:{Password}@{Server_IP}:5432/rin-deviantart-tokens"
+            f"postgresql+asyncpg://{Username}:{Password}@{Server_IP}:5432/rin_deviantart_tokens_v2"
         )
         tokens = Table(
             "DA_Tokens",
-- 
2.35.1

